AlfrescoDocumentPickerFileProvider/FileProvider.m

```diff
@@ -492,4 +478,230 @@ - (nullable NSFileProviderItemIdentifier)persistentIdentifierForItemAtURL:(NSURL
+     }];
+}
+
+- (void)prepareFileUploadForEditedItemAtURL:(NSURL *)url
+                             destinationURL:(NSURL *)destinationURL
+                        withCompletionBlock:(void (^)(NSURL *readingURL, NSError *error))completionBlock
+{
+    __block NSError *error = nil;
+    __weak typeof(self) weakSelf = self;
+    [self.fileCoordinator coordinateReadingItemAtURL:url
+                                             options:NSFileCoordinatorReadingForUploading
+                                               error:&error
+                                          byAccessor:^(NSURL *newReadingURL)
+     {
+         __strong typeof(self) strongSelf = weakSelf;
+         if (error) {
+             completionBlock(nil, error);
+             return ;
+         }
+         
+         [strongSelf.fileCoordinator coordinateWritingItemAtURL:destinationURL
+                                                        options:NSFileCoordinatorWritingForReplacing
+                                                          error:&error
+                                                     byAccessor:^(NSURL * _Nonnull newURL)
+          {
+              if (error) {
+                  completionBlock(nil, error);
+                  return;
+              } else {
+                  [weakSelf.fileService saveDocumentAtURL:newReadingURL
+                                                    toURL:newURL
+                                  overwritingExistingFile:YES];
+                  
+                  completionBlock(newReadingURL, nil);
+              }
+          }];
+     }];
+}
+
+- (void)uploadEditedItem:(AlfrescoDocument *)alfrescoDocument
+      withItemIdentifier:(NSFileProviderItemIdentifier)itemIdentifier
+                   atURL:(NSURL *)url
+    forAccountIdentifier:(NSString *)accountIdentifier
+
+{
+    __block AlfrescoDocument *document = alfrescoDocument;
+    __block BOOL networkOperationCallbackComplete = NO;
+    __weak typeof(self) weakSelf = self;
+    
+    // Session exists, use that, else do a login and then upload.
+    [self.accountManager getSessionForAccountIdentifier:accountIdentifier
+                                      networkIdentifier:nil
+                                    withCompletionBlock:^(id<AlfrescoSession> session, NSError *loginError)
+     {
+         if(session)
+         {
+             [weakSelf uploadDocument:document
+                            sourceURL:url
+                              session:session
+                      completionBlock:^(AlfrescoDocument *document, NSError *updateError) {
+                          if (updateError)
+                          {
+                              AlfrescoLogError(@"Error Updating Document: %@. Error: %@", document.name, updateError.localizedDescription);
+                          }
+                          else
+                          {
+                              [[AFPDataManager sharedManager] updateSyncDocument:document
```

Comment:
The name of at least one of these variables should be changed for better clarity.